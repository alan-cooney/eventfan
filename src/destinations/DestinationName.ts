/* eslint-disable import/prefer-default-export */

/**
 * Destination names
 */
export enum DestinationName {
  FACEBOOK_PIXEL = "FACEBOOK_PIXEL",
  HOTJAR = "HOTJAR",
  GA4 = "GA4",
  POSTHOG = "POSTHOG",
}
