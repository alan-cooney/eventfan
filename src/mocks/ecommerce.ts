import {
  CartShared,
  CartViewed,
  CheckoutStarted,
  CheckoutStepCompleted,
  CheckoutStepViewed,
  OrderCompleted,
  OrderRefunded,
  PaymentInfoEntered,
  ProductAdded,
  ProductClicked,
  ProductListViewed,
  ProductRemoved,
  ProductShared,
  ProductsSearched,
  ProductViewed,
  PromotionClicked,
  PromotionViewed,
} from "../sourceEvents/ecommerce";

/**
 * Helper Mock Product
 */
const mockProduct: ProductViewed["properties"] = {
  category: "Trip",
  name: "Product Name",
  price: 100.0,
  product_id: "productID",
  quantity: 1,
  sku: "sku",
  variant: "variant",
};

export const mockCartShared: CartShared = {
  name: "Cart Shared",
  properties: {
    cart_id: "cartID",
    products: [mockProduct],
    recipient: "recipient",
    share_message: "share message",
    share_via: "Facebook",
  },
};

export const mockCartViewed: CartViewed = {
  name: "Cart Viewed",
  properties: {
    cart_id: "cartID",
    products: [mockProduct],
  },
};

export const mockCheckoutStarted: CheckoutStarted = {
  name: "Checkout Started",
  properties: {
    currency: "GBP",
    order_id: "orderID",
    products: [mockProduct],
    revenue: 100.0,
    total: 200.0,
  },
};

export const mockCheckoutStepCompleted: CheckoutStepCompleted = {
  name: "Checkout Step Completed",
  properties: {
    checkout_id: "checkoutID",
    payment_method: "Credit Card",
    shipping_method: "shippingMethod",
    step: 1,
  },
};

export const mockCheckoutStepViewed: CheckoutStepViewed = {
  name: "Checkout Step Viewed",
  properties: mockCheckoutStepCompleted.properties,
};

export const mockOrderCompleted: OrderCompleted = {
  name: "Order Completed",
  properties: mockCheckoutStarted.properties,
};

export const mockOrderRefunded: OrderRefunded = {
  name: "Order Refunded",
  properties: mockCheckoutStarted.properties,
};

export const mockPaymentInfoEntered: PaymentInfoEntered = {
  name: "Payment Info Entered",
  properties: {
    currency: "GBP",
    order_id: "orderID",
    products: [mockProduct],
    revenue: 100.0,
  },
};

export const mockProductAdded: ProductAdded = {
  name: "Product Added",
  properties: mockProduct,
};

export const mockProductClicked: ProductClicked = {
  name: "Product Clicked",
  properties: mockProduct,
};

export const mockProductListViewed: ProductListViewed = {
  name: "Product List Viewed",
  properties: {
    category: "category",
    list_id: "list_id",
    products: [mockProduct],
  },
};

export const mockProductRemoved: ProductRemoved = {
  name: "Product Removed",
  properties: mockProduct,
};

export const mockProductsShared: ProductShared = {
  name: "Product Shared",
  properties: {
    ...mockProduct,
    recipient: "recipient",
    share_message: "share message",
    share_via: "Facebook",
  },
};

export const mockProductsSearched: ProductsSearched = {
  name: "Products Searched",
  properties: {
    query: "Query string",
  },
};

export const mockProductViewed: ProductViewed = {
  name: "Product Viewed",
  properties: mockProduct,
};

export const mockPromotionClicked: PromotionClicked = {
  name: "Promotion Clicked",
  properties: {
    promotion_creative: "creative",
    promotion_id: "promotionID",
    promotion_name: "name",
    promotion_position: "position",
  },
};

export const mockPromotionViewed: PromotionViewed = {
  name: "Promotion Viewed",
  properties: mockPromotionClicked.properties,
};
