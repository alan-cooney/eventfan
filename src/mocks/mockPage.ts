import { Page } from "../types/PageViewProps";

const mockPage: Page = {
  options: undefined,
  name: "title",
  properties: {
    path: "/path",
    title: "title",
    url: "https://www.example.com/path",
  },
};

export default mockPage;
