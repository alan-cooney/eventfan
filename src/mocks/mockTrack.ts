import { TEvent } from "../types/TrackEvent";

const mockTrack: TEvent = {
  name: "Test Event",
  properties: {
    iceCream: "vanilla",
  },
};

export default mockTrack;
