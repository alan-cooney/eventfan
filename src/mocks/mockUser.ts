import { User } from "../types/User";

const mockUser: User = {
  options: undefined,
  traits: {
    email: "test@gmail.com",
    firstName: "First Name",
  },
  userId: "userID",
};

export default mockUser;
