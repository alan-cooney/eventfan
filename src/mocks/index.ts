import mockPage from "./mockPage";
import mockTrack from "./mockTrack";
import mockUser from "./mockUser";

export { mockPage, mockTrack, mockUser };
export * from "./ecommerce";
