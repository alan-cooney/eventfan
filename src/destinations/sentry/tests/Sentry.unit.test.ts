import { mockOrderCompleted, mockUser } from "../../../mocks";
import { DestinationName } from "../../DestinationName";
import Sentry from "../Sentry";

it("has the correct destination name", () => {
  const destination = new Sentry({});
  expect(destination.name).toBe(DestinationName.SENTRY);
});

it("starts as not loaded", () => {
  const destination = new Sentry({});
  expect(destination.isLoaded).toBeFalsy();
});

describe("identify", () => {
  it("passes on the traits to the .setUser method", async () => {
    const setUser = jest.fn();
    const addBreadcrumb = jest.fn();
    const sentry = new Sentry({ setUser, addBreadcrumb });

    await sentry.identify(mockUser);

    expect(setUser).toHaveBeenCalled();
    expect(setUser.mock.calls[0]).toMatchSnapshot();
  });
});

describe("track", () => {
  it("creates a breadcrumb matching the snapshot", async () => {
    const setUser = jest.fn();
    const addBreadcrumb = jest.fn();
    const sentry = new Sentry({ setUser, addBreadcrumb });

    await sentry.track({
      name: "Purchase",
      properties: mockOrderCompleted.properties,
    });

    expect(addBreadcrumb).toHaveBeenCalled();
    expect(addBreadcrumb.mock.calls[0]).toMatchSnapshot();
  });
});
