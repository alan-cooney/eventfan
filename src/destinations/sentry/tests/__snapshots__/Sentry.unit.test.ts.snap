// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`identify passes on the traits to the .setUser method 1`] = `
[
  {
    "address": {
      "city": "City",
      "country": "GB",
      "postalCode": "SW11 111",
      "state": "London",
      "street": "1 Road Name",
    },
    "age": 30,
    "birthday": "1990-01-01",
    "createdAt": 2020-01-01T00:00:00.000Z,
    "email": "test@gmail.com",
    "firstName": "FirstName",
    "gender": "MALE",
    "id": "userID",
    "lastName": "LastName",
    "name": "FirstName LastName",
    "phone": "+447111111111",
    "tags": [
      "admin",
    ],
    "username": "userID",
  },
]
`;

exports[`track creates a breadcrumb matching the snapshot 1`] = `
[
  {
    "category": "Purchase",
    "data": {
      "currency": "GBP",
      "order_id": "orderID",
      "products": [
        {
          "category": "Trip",
          "currency": "GBP",
          "image_url": "http://example.com/image.jpg",
          "name": "Product Name",
          "price": 100,
          "product_id": "productID",
          "quantity": 1,
          "sku": "sku",
          "url": "http://example.com/productID",
          "variant": "variant",
        },
      ],
      "revenue": 100,
      "total": 200,
    },
    "level": "info",
  },
]
`;
